mycroft/audio/service.py

```diff
@@ -0,0 +1,224 @@
+
+        # update TTS object if configuration has changed
+        if not self._tts_hash or self._tts_hash != config.get("module", ""):
+            if self.tts:
+                self.tts.shutdown()
+            # Create new tts instance
+            LOG.info("(re)loading TTS engine")
+            self.tts = TTSFactory.create(config)
+            self.tts.init(self.bus)
+            self._tts_hash = config.get("module", "")
+
+        # if fallback TTS is the same as main TTS dont load it
+        if config.get("module", "") == config.get("fallback_module", ""):
+            return
+
+        if not self._fallback_tts_hash or \
+                self._fallback_tts_hash != config.get("fallback_module", ""):
+            if self.fallback_tts:
+                self.fallback_tts.shutdown()
+            # Create new tts instance
+            LOG.info("(re)loading fallback TTS engine")
+            self._get_tts_fallback()
+            self._fallback_tts_hash = config.get("fallback_module", "")
+
+    def mute_and_speak(self, utterance, ident, listen=False):
+        """Mute mic and start speaking the utterance using selected tts backend.
+
+        Args:
+            utterance:  The sentence to be spoken
+            ident:      Ident tying the utterance to the source query
+        """
+        LOG.info("Speak: " + utterance)
+        try:
+            self.tts.execute(utterance, ident, listen)
+        except Exception as e:
+            LOG.exception(f"TTS synth failed! {e}")
+            if self._tts_hash != self._fallback_tts_hash:
+                self.execute_fallback_tts(utterance, ident, listen)
+
+    def _get_tts_fallback(self):
+        """Lazily initializes the fallback TTS if needed."""
+        if not self.fallback_tts:
+            config = Configuration()
+            engine = config.get('tts', {}).get("fallback_module", "mimic")
+            cfg = {"tts": {"module": engine,
+                           engine: config.get('tts', {}).get(engine, {})}}
+            self.fallback_tts = TTSFactory.create(cfg)
+            self.fallback_tts.validator.validate()
+            self.fallback_tts.init(self.bus)
+
+        return self.fallback_tts
+
+    def execute_fallback_tts(self, utterance, ident, listen):
+        """Speak utterance using fallback TTS if connection is lost.
+
+        Args:
+            utterance (str): sentence to speak
+            ident (str): interaction id for metrics
+            listen (bool): True if interaction should end with mycroft listening
+        """
+        try:
+            self.tts = self._get_tts_fallback()
+            LOG.debug("TTS fallback, utterance : " + str(utterance))
+            self.tts.execute(utterance, ident, listen)
+            return
+        except Exception as e:
+            LOG.error(e)
+            LOG.exception(f"TTS FAILURE! utterance : {utterance}")
+
+    def mimic_fallback_tts(self, utterance, ident, listen):
+        """
+        DEPRECATED: use execute_fallback_tts instead
```

Comment:
can this method be deleted?

Reply:
Probably; backwards-compat now just emits a `speak` event